src/algorithms/all_neighbors.c

```diff
@@ -15,15 +15,14 @@ static void _AllNeighborsCtx_CollectNeighbors
 	EntityID id
 ) {
 	ctx->current_level++;
-	GxB_MatrixTupleIter *iter;
 	if(ctx->current_level == array_len(ctx->levels)) {
-		GxB_MatrixTupleIter_new(&iter, ctx->M);
+		ASSERT(ctx->M);
```

Comment:
do not perform this test here, should be performed on both `AllNeighborsCtx_New` and `AllNeighborsCtx_Reset`